### ISSUE TITLE
Create Pattern Formatter for better human output of complex patterns

### ISSUE THREAD

**Issue by declension** [quodlibet/quodlibet#1765]
Following on from #1762

Currently (AIUI) it _evaluates_ the pattern over a dummy `FakeSong` object which returns human-readable names instead of tag values, with slightly different formatting on multiple.

Whilst this works for simple expressions, any conditionals get evaluated (to false, mostly) and are not expressed. 

Either some kind of semi-verbatim output might be good, or even better an English like description e.g.

    <albumartist|<albumartist>|<artist>> 
becomes _Album Artist (else Artist)_

Full of complexity but some best efforts might be better than the current solution

**Comment by lazka** [comment]
The idea behind the current approach (no taking into account the new query conditions) is that usually the "else" part is a fallback and the "if" part is the preferred output, so the title displays the preferred one. And I couldn't think of anything better..

**Comment by declension** [comment]
Yep, and that seems like a good default. What I forgot to mention was the original case of
```"<#(bitrate \> 150)|HQ|LQ>"), "LQ")```

generates a column header of _LQ_ which is not very useful here, so I guess an additional feature should be some kind of parsing of numeric expressions too... In the ideal case this might say a Pythonic _HQ if #bitrate >150 else LQ_ but this is pretty verbose and YMMV etc (Just _LQ|HQ_ would be a good start

**Comment by lazka** [comment]
What about allowing `<condition|title|then|else>`?

**Comment by lazka** [comment]
Or `<condition|then|else|title>` because that makes parsing easier
